### PR TITLE
[FIX] stock: round w/ `precision_rounding`

### PR DESCRIPTION
**Pull request by lvsz**
Versions
--------
- 17.0+

Issue
-----
`stock.quant`'s `_get_reserve_quantity` method makes a call to `float_compare`, passing a `precision_rounding` value as a `precision_digits` parameter.

Issue found working on opw-3705546

### CODE REVIEW

**Review by HANNICHE-Walid — APPROVED** [review]
LGTM

File: addons/stock/tests/test_quant.py [hunk 1]
```diff
@@ -1116,6 +1116,20 @@ def test_unpack_and_quants_history(self):
         history = self.env['stock.move.line'].search(action['domain'])
         self.assertTrue(history)
 
+    def test_reserve_fractional_qty(self):
+        self.env['stock.quant']\
+            ._update_available_quantity(self.product_serial, self.stock_location, 1.1)
```
Comment (Sehnde) [inline addons/stock/tests/test_quant.py:1121]:
```suggestion
        lot1 = self.env['stock.lot'].create({'name': 'lot1', 'product_id': self.product_serial.id})
        lot2 = self.env['stock.lot'].create({'name': 'lot2', 'product_id': self.product_serial.id})
        for lot in (lot1, lot2):
            self.env['stock.quant']._update_available_quantity(
                product_id=self.product_serial,
                location_id=self.stock_location,
                quantity=1,
                lot_id=lot)
```

File: addons/stock/tests/test_quant.py [hunk 2]
```diff
@@ -1116,6 +1116,20 @@ def test_unpack_and_quants_history(self):
...
+        move = self.env['stock.move'].create({
+            'name': 'test_reserve_small_qty',
+            'location_id': self.stock_location.id,
+            'location_dest_id': self.stock_subloc2.id,
+            'product_id': self.product_serial.id,
+            'product_uom_qty': 2.0,
```
Comment (Sehnde) [inline addons/stock/tests/test_quant.py:1127]:
```suggestion
            'product_uom_qty': 1.1,
```

**Review by Sehnde — CHANGES_REQUESTED** [review]
Fix is ok but the test should be changed to fit a 'possible real situation'. By that I mean that it is more likely that a user will try to reserve a float amount for a serial product than having float amount of a serial in stock.
This is nitpicking but seeing a serial with quantity of 1.1 hurts my feelings.

**Review by Sehnde — APPROVED** [review]

**Comment by kitan191** [comment]
@robodoo delegate+

**Comment by lvsz** [comment]
@robodoo r+

**Pull request closed without merging**